### PR TITLE
[WIP] Softcut reset

### PR DESCRIPTION
**Pull request by catfact**
another totally untested PR for issue #812 

depends on PR #854, sorry

### CODE REVIEW

File: matron/src/weaver.c [hunk 1]
```diff
@@ -292,6 +293,7 @@ void w_init(void) {
   lua_register_norns("cut_buffer_read_stereo", &_cut_buffer_read_stereo);
   lua_register_norns("cut_buffer_write_mono", &_cut_buffer_write_mono);
   lua_register_norns("cut_buffer_write_stereo", &_cut_buffer_write_stereo);
+  lua_register_norns("cut_buffer_reset", &_cut_reset);
```
Comment (tehn) [inline matron/src/weaver.c]: no `buffer` right?

```
lua_register_norns("cut_reset", &_cut_reset);
```

**Comment by tehn** [comment]
ok i will test this along with the other PR. thank you tons

File: matron/src/weaver.c (hunk 1 above)
Comment (catfact) [reply matron/src/weaver.c]: Woop

**Comment by catfact** [comment]
I can  get to testing both a little later this evening 

I also want to try the worker threads for buf read / write, again

If you have time to make a test script I will happily use it

**Comment by catfact** [comment]
sorry i have screwed up and made fixes to the filter branch on this one, its a mess to separate them now.

i'm going to close both PRs and just open a combined one.

the changes are not hard to follow

sorry sorry
